libraries/provider_install_source.rb

```diff
@@ -0,0 +1,68 @@
+class Chef
+  # Chef Provider for installing or removing Elasticsearch from package or source
+  class Provider::ElasticsearchInstallSource < Provider
+    include Poise
+    include ElasticsearchCookbook::Helpers
+    if respond_to?(:provides)
+      provides :elasticsearch_install
+    end
+
+    def action_install
+      converge_by("#{new_resource.name} - install source") do
+        notifying_block do
+          include_recipe 'ark'
```

Comment:
My goal here was to re-use as much as possible at the start. We could definitely re-implement it in pure ruby :) I've written some initial tests that should help us be sure we don't introduce bugs on a re-write, but we'd definitely want to do more tests if we started manipulating the tarball directly. :+1: